aws-ssm-opsmetadata/aws-ssm-opsmetadata.json

```diff
@@ -0,0 +1,127 @@
+            "properties": {
+                "Key": {
+                    "description": "The name of the tag.",
+                    "type": "string",
+                    "pattern": "^([\\p{L}\\p{Z}\\p{N}_.:/=+\\-@]*)$",
+                    "minLength": 1,
+                    "maxLength": 128
+                },
+                "Value": {
+                    "description": "The value of the tag.",
+                    "type": "string",
+                    "pattern": "^([\\p{L}\\p{Z}\\p{N}_.:/=+\\-@]*)$",
+                    "minLength": 1,
+                    "maxLength": 256
+                }
+            },
+            "required": [
+                "Value",
+                "Key"
+            ],
+            "additionalProperties": false
+        }
+    },
+    "properties": {
+        "ResourceId": {
+            "description": "Resource Id for which OpsMetadata is being created.",
+            "type": "string",
+            "pattern": "^(?!\\s*$).+"
+        },
+        "Metadata": {
+            "description": "A map of metadata key-value pairs to be stored in the OpsMetadata object.",
+            "type": "object",
+            "patternProperties": {
+                ".*{1,255}": {
+                    "$ref": "#/definitions/MetadataValue"
+                }
+            },
+            "additionalProperties": false
+        },
+        "Tags": {
+            "description": "An array of tag key-value pairs to apply to this resource.",
+            "type": "array",
+            "uniqueItems": true,
+            "insertionOrder": false,
+            "items": {
+                "$ref": "#/definitions/Tag"
+            }
+        },
+        "OpsMetadataArn": {
+            "description": "Arn of the OpsMetadata Object",
+            "type": "string",
+            "pattern": "arn:(aws[a-zA-Z-]*)?:ssm:[a-z0-9-\\.]{0,63}:[a-z0-9-\\.]{0,63}:opsmetadata/([a-zA-Z0-9-_\\./]*)"
+        }
+    },
+    "additionalProperties": false,
+    "required": [
+        "ResourceId"
+    ],
+    "readOnlyProperties": [
+        "/properties/OpsMetadataArn"
+    ],
+    "primaryIdentifier": [
+        "/properties/OpsMetadataArn"
+    ],
+    "createOnlyProperties": [
+        "/properties/ResourceId"
+    ],
+    "handlers": {
+        "create": {
+            "permissions": [
+                "ssm:CreateOpsMetadata",
+                "ssm:GetOpsMetadata"
+            ],
+            "timeoutInMinutes": 5
+        },
+        "read": {
+            "permissions": [
+                "ssm:GetOpsMetadata"
+            ]
+        },
+        "update": {
+            "permissions": [
+                "ssm:CreateOpsMetadata",
+                "ssm:AddTagsToResource",
+                "ssm:RemoveTagsFromResource",
+                "ssm:ListTagsForResource",
+                "ssm:GetOpsMetadata"
+            ],
+            "timeoutInMinutes": 5
+        },
+        "delete": {
+            "permissions": [
+                "ssm:DeleteOpsMetadata"
```

Comment:
should this include `ssm:RemoveTagsFromResource` as well?

Reply:
It is not required as the service takes care of deleting the tags.